### PR TITLE
Feature/star 254

### PR DESCRIPTION
**Pull request by KrzysztofDziedziechEscolasoft**

### CODE REVIEW

File: src/EscolaLmsInvoicesServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    public function register()
+    {
+        parent::register();
+    }
+
+    protected function bootForConsole(): void
+    {
+        $this->publishes([
+            __DIR__ . '/config.php' => config_path('invoices.php'),
```
Comment (pa-cholek) [inline src/EscolaLmsInvoicesServiceProvider.php:37]: If you are publishing some config file then you should also load it / merge with currently loaded config in `register()`

```
    $this->mergeConfigFrom(
    __DIR__ . '/config.php', 'escolalms_invoices'
    );
    ```

**Review by pa-cholek — APPROVED** [review]

File: src/Http/Controllers/InvoicesApiController.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace EscolaLms\Invoices\Http\Controllers;
+
+use EscolaLms\Core\Http\Controllers\EscolaLmsBaseController;
+use EscolaLms\Invoices\Http\Controllers\Swagger\InvoicesApiContract;
+use EscolaLms\Invoices\Http\Requests\InvoicesReadRequest;
+use EscolaLms\Invoices\Services\Contracts\InvoicesServiceContract;
+use Exception;
+use Illuminate\Http\Response;
```
Comment (HerbertIV) [inline src/Http/Controllers/InvoicesApiController.php:10]: Remove unused classes

File: src/Http/Requests/InvoicesReadRequest.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+namespace EscolaLms\Invoices\Http\Requests;
+
+use EscolaLms\Cart\Models\Order;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Gate;
+use Illuminate\Validation\Rule;
+
+class InvoicesReadRequest extends FormRequest
+{
+    public function authorize()
```
Comment (HerbertIV) [inline src/Http/Requests/InvoicesReadRequest.php]:
```suggestion
    public function authorize(): bool
```

File: src/Services/InvoicesService.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            ->logo(public_path(Config::get('invoices.logo') ?? 'vendor/invoices/sample-logo.png'));
+
+        return $invoice;
+    }
+
+    private function prepareClient(): Party
+    {
+        return new Party(Config::get('invoices.seller') ?? [
+            'name' => 'Escola',
+        ]);
```
Comment (HerbertIV) [inline src/Services/InvoicesService.php]:
```suggestion
        return new Party(Config::get('invoices.seller',  [
            'name' => 'Escola',
        ]));
```

File: tests/Models/Course.php [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    protected static function newFactory(): CourseFactory
+    {
+        return CartCourseFactory::new();
+    }
+
+    public function getBuyableDescription()
+    {
+        return 'abc test';
```
Comment (HerbertIV) [inline tests/Models/Course.php:91]:
```suggestion
        return '';
```

**Review by HerbertIV — APPROVED** [review]

File: tests/Models/Course.php (hunk 5 above)
Comment (KrzysztofDziedziechEscolasoft) [reply tests/Models/Course.php:91]: this is only for tests

File: src/EscolaLmsInvoicesServiceProvider.php (hunk 1 above)
Comment (KrzysztofDziedziechEscolasoft) [reply src/EscolaLmsInvoicesServiceProvider.php:37]: done

File: src/Services/InvoicesService.php (hunk 4 above)
Comment (KrzysztofDziedziechEscolasoft) [reply src/Services/InvoicesService.php]: done

File: src/Http/Controllers/InvoicesApiController.php (hunk 2 above)
Comment (KrzysztofDziedziechEscolasoft) [reply src/Http/Controllers/InvoicesApiController.php:10]: done

**Review by qunabu — APPROVED** [review]

**Pull request merged**